lockrenewal.go

```diff
@@ -0,0 +1,113 @@
+package servicebus
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-amqp-common-go/log"
+	"github.com/Azure/azure-amqp-common-go/uuid"
+	otlogger "github.com/opentracing/opentracing-go/log"
+
+	"pack.ag/amqp"
+)
+
+const (
+	serviceBuslockRenewalOperationName = "com.microsoft:renew-lock"
+)
+
+func (e *entity) createManagementChannels(ctx context.Context, amqpPath string) (*sender, *receiver, error) {
+	if e.namespace == nil {
+		panic("expect namespace not nil")
+	}
+
+	subscriptionAddress := e.namespace.getEntityManagementPath(amqpPath)
+
+	// receiver := uuid.NewV4().String()
+	sender, err := e.namespace.newSender(ctx, subscriptionAddress)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	reciever, err := e.namespace.newReceiver(ctx, subscriptionAddress)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return sender, reciever, nil
+}
+
+//RenewLocks renews the locks on messages provided
+func (e *entity) RenewLocks(ctx context.Context, messages []*Message) error {
+	lockTokens := make([]*uuid.UUID, 0, len(messages))
+	for _, m := range messages {
+		if m.LockToken == nil {
+			log.For(ctx).Error(fmt.Errorf("failed: message has nil lock token, cannot renew lock"), otlogger.Object("messageId", m))
+			continue
+		}
+
+		lockTokens = append(lockTokens, m.LockToken)
+	}
+
+	if len(lockTokens) < 1 {
+		log.For(ctx).Info("no lock tokens present to renew")
+		return nil
+	}
+
+	e.renewMessageLockMutex.Lock()
+	defer e.renewMessageLockMutex.Unlock()
+
+	sender, receiver, err := e.createManagementChannels(ctx, e.Name)
```

Comment:
Check out the RPC stuff here: https://github.com/Azure/azure-amqp-common-go/blob/master/rpc/rpc.go.

The RPC functionality is used for Event Hubs management requests as seen here: https://github.com/Azure/azure-event-hubs-go/blob/master/amqp_mgmt.go.

I think this would reduce your code a bit.